libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3605,8 +3642,8 @@ void GCS_MAVLINK::handle_common_message(const mavlink_message_t &msg)
         break;
 
     case MAVLINK_MSG_ID_REQUEST_DATA_STREAM:
-        // only pass if override is not selected 
-        if (!(_port->get_options() & _port->OPTION_NOSTREAMOVERRIDE)) {
+        // only pass if override is not selected or high latency
+        if (!(_port->get_options() & _port->OPTION_NOSTREAMOVERRIDE) || !is_high_latency_link) {
```

Comment:
Doesn't this line mean you *can't* request `HIGH_LATENCY2` at a higher rate?

I suggest that users should be permitted to request messages if they really feel like it over high-latency links; they already have the NOSTREAMOVERRIDE option to save them if that would be a disaster.

Reply:
> Doesn't this line mean you can't request HIGH_LATENCY2 at a higher rate?

You can via ``MAV_CMD_SET_MESSAGE_INTERVAL``

> I suggest that users should be permitted to request messages if they really feel like it over high-latency links; they already have the NOSTREAMOVERRIDE option to save them if that would be a disaster.

This only disables the streamrates. ``MAV_CMD_SET_MESSAGE_INTERVAL`` is still available.

If the user wants streamrates, they can use the SerialProtocol_MAVLink2 :)